api/groups/baseGroup.go

```diff
@@ -59,15 +69,61 @@ func (bg *baseGroup) RemoveEndpoint(path string) error {
 }
 
 // RegisterRoutes will register all the endpoints to the given web server
-func (bg *baseGroup) RegisterRoutes(ws *gin.RouterGroup) {
+func (bg *baseGroup) RegisterRoutes(ws *gin.RouterGroup, apiConfig data.ApiRoutesConfig, authenticationFunc gin.HandlerFunc) {
```

Comment:
add unit tests for this new functionality

Reply:
it is already tested in all groups unit tests. further testing (check gin requests/responses) is not possible as for now.